x/tokenfactory/keeper/createdenom.go

```diff
@@ -49,6 +52,16 @@ func (k Keeper) createDenomAfterValidation(ctx sdk.Context, creatorAddr, denom s
 		return err
 	}
 
+	// create erc20 contractAddr and set token pair
+	ethContractAddr := ethcommon.BytesToAddress([]byte(denom))
```

Comment:
different denoms like `factory/mantra1x7x9pkfxf33l87ftspk5aetwnkr0lvlvdy9gff/test` and `factory/mantra1x8x9pkfxf33l87ftspk5aetwnkr0lvlvdy9gff/test` could conflict when have the same last 20 bytes?